### ISSUE TITLE
Setting up org-ref with ivy-bibtex instead of helm-bibtex

### ISSUE THREAD

**Issue by tim-hilt** [jkitchin/org-ref#793]
It could be, that this question has already a definitive answer elsewhere, but since i couldn't find it here's my issue:

I installed org-ref and got it working as i intended it to. The only thing that really annoys me is [the dependency on helm](https://melpa.org/#/org-ref). I use ivy throughout my setup and would love to keep things consistent.

I found [org-ref-ivy](https://github.com/yyadavalli/org-ref-ivy) but the last commit was two years before and i'm not good enough in writing elisp (yet!) to look at the code and understand what i would have to do.

Is there some way to get org-ref working without having to install helm but to use ivy and ivy-bibtex?

@jkitchin from my point of view the issue can be closed as soon as someone delivers an answer even though it isn't a solution to my problem. Thanks for making this package happen! I would happily use it with helm if there's no other possibility!

**Comment by jkitchin** [comment]
There are ivy backends in org-ref already. This one doesn't use ivy-bibtex, but one that I designed for  org-ref.

```
(setq org-ref-completion-library 'org-ref-ivy-cite)
(require 'org-ref)
```

There is probably a way to use ivy-bibtex, you just have to set variables like org-ref-insert-cite-function to a function that uses ivy-bibtex for that. I have never used ivy-bibtex, so I don't have a sense for how easy it would be to integrate. I guess not hard.

the helm dependency is a feature, even if you don't use helm. It turns out to be too difficulty to separate org-ref into several packages, e.g. a core and the different backends. I tried many years ago, but was unsuccessful an am not interested in trying it again.
